fabric-data-attachment-api-v1/src/main/java/net/fabricmc/fabric/api/attachment/v1/AttachmentRegistry.java

```diff
@@ -0,0 +1,139 @@
+		Objects.requireNonNull(id, "identifier cannot be null");
+		Objects.requireNonNull(defaultValue, "default value cannot be null");
+
+		return AttachmentRegistry.<A>builder().defaultValue(defaultValue).buildAndRegister(id);
+	}
+
+	/**
+	 * Creates <i>and registers</i> an attachment, initialized using a default value, and optionally a {@link AttachmentSerializer serializer}.
+	 *
+	 * @param id the identifier of this attachment
+	 * @param defaultValue the default value, used for objects which don't have attached data yet. Must not be {@code null}
+	 * @param serializer the serializer for this attachment. {@code null} for no serializer
+	 * @param <A> the type of attached data
+	 * @return the registered {@link Attachment} instance
+	 */
+	public static <A> Attachment<A> createDefaulted(Identifier id, A defaultValue, AttachmentSerializer<A> serializer) {
+		Objects.requireNonNull(id, "identifier cannot be null");
+		Objects.requireNonNull(defaultValue, "default value cannot be null");
+		Objects.requireNonNull(serializer, "serializer cannot be null");
+
+		return AttachmentRegistry.<A>builder().defaultValue(defaultValue).buildAndRegister(id);
+	}
+
+	/**
+	 * Gets the {@link Attachment} associated with the given {@link Identifier}, or {@code null} if none was registered.
+	 *
+	 * @param id the identifier of the attachment
+	 * @return the corresponding attachment, or {@code null} if it doesn't exist
+	 */
+	@Nullable
+	public static Attachment<?> get(Identifier id) {
+		return AttachmentRegistryImpl.get(id);
+	}
+
+	/**
+	 * Creates a {@link Builder}, that gives finer control over the attachment's properties.
+	 *
+	 * @param <A> the type of the attached data
+	 * @return a {@link Builder} instance
+	 */
+	public static <A> Builder<A> builder() {
+		return AttachmentRegistryImpl.builder();
+	}
+
+	/**
+	 * A builder for creating {@link Attachment}s with finer control over their properties.
+	 *
+	 * @param <A> the type of the attached data
+	 */
+	public interface Builder<A> {
+		/**
+		 * Sets whether the attachment should be synced between server and client or not.
+		 *
+		 * @param synced whether the attachment data should be synced
+		 * @return the builder
+		 */
+		Builder<A> synced(boolean synced);
+
+		/**
+		 * Makes the attachment use the argument as a starting value for objects which don't have attached data yet.
+		 *
+		 * @param defaultVal the default value
+		 * @return the builder
+		 */
+		Builder<A> defaultValue(A defaultVal);
+
+		/**
+		 * Uses the given {@link Supplier} to get a starting value for objects which don't have attached data yet.
+		 *
+		 * @param supplier the initializer
+		 * @return the builder
+		 */
+		Builder<A> initializer(Supplier<A> supplier);
+
+		/**
+		 * Sets the {@link AttachmentSerializer} for this attachment.
+		 *
+		 * @param serializer the serializer
+		 * @return the builder
+		 */
+		Builder<A> serializer(AttachmentSerializer<A> serializer);
```

Comment:
Can we have here a simplification for codecs like this?
```java
default Builder<A> serializer(Codec<A> codec) {
    return serializer(AttachmentSerializer.fromCodec(codec));
}
```
In fact, I don't see the point in giving anything other than codecs, but I understand that some people don't like codecs. Although I think modders should just get used to codecs since they'll be everywhere in the future

Reply:
I agree, with the way Mojang are changing seemingly everything over to Codecs, we should do the same with new APIs.

Reply:
I can certainly add shortcuts for Codecs, but should I go further and ditch AttachmentSerializer in favor of them?

Reply:
I dont see why not?

Reply:
Yes, that's exactly what I meant

Reply:
I guess the old interface design was more appropriate for the old proposal's receiver-aware design, now there's nothing this can do than codecs can't.